src/Controller/Admin/Asset/AssetController.php

```diff
@@ -1241,6 +1241,13 @@ public function downloadImageThumbnailAction(Request $request): BinaryFileRespon
         }
 
         if ($thumbnail) {
+            $thumbnailConfig = $thumbnail->getConfig();
+            if ($thumbnailConfig->getFormat() === 'SOURCE' &&
+                $autoFromatConfigs = $thumbnailConfig->getAutoFormatThumbnailConfigs()) {
```

Comment:
`getAutoFormatThumbnailConfigs()` is internal

Reply:
is there any drawback? i see that the controller itself and other methods used in here are internal too

Reply:
We should remove the internal flag if the method is used outside of the repository.
So it can't be changed without BC break, because it is used external.

Reply:
I think that's a good point, looks like it may potentially and inadvertently change the expected output and be harder to maintain, but don't we just have to make sure that any change to `getAutoFormatThumbnailConfigs` would have the right follow-up to all its usage and "internally" treat it as if something that can bc-break? 🤔 
By removing the internal flag, it kind of "invites" the devs to use it in their own scripts, which doesn't seem a purpose for this method

Reply:
@kingjia90 When you need to use it external, why should be a dev not use it external? When you use internal methods in other repositories it will be hard to maintain. Because every change on internal methods you must search in all repositories for usages. Do you realy want handle this? Maybe this logic should be in the core and you make a new external method for this repository and for all devs?

Reply:
this bundle is still tightly coupled with the core as it still uses lot of core internal classes so we can go ahead with this change.